### PR TITLE
LPS-74849

### PR DESCRIPTION
**Pull request by hhuijser**

### CODE REVIEW

**Comment by pyoo47** [comment]
ci:retest

**Comment by hhuijser** [comment]
@shuyangzhou, do you know why this if failing?
I added the kernel dependency to `modules/util/portal-tools-target-platform-indexer-client/build.gradle`

File: modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/ServiceTrackerUtil.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 package com.liferay.osgi.service.tracker.collections.internal;
 
+import com.liferay.portal.kernel.util.StringBundler;
```
Comment (shuyangzhou) [inline modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/ServiceTrackerUtil.java]: This is the cause of the failure. osgi-service-tracker-collections does not have kernel dependencies.
You will have to use petra-string's StringBundler here, and add petra-string's dependencies.
As a matter of fact, @tomwang2011 will be replacing globally to move all kernel StringBundler's usage to petra-string's StringBundler.
You can either update all usages to that, or just do it for osgi-service-tracker-collections, leaving the rest to @tomwang2011

**Comment by hhuijser** [comment]
@shuyangzhou, thanks!
I'll probably just fix this one and let Tom do this thing...

**Pull request closed without merging**